### PR TITLE
Update and refactor requirements.txt

### PR DESCRIPTION
**Pull request by riedgar-ms**
Split requirements.txt into sections
Add missing dependencies

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+autopep8
+
+# Required for test
+# Pin pytest due to VS Code issue
+pytest==5.0.1
+scikit-learn
+
+# Required for notebooks
+jupyter
+matplotlib
```
Comment (romanlutz) [inline requirements.txt]: matplotlib is actually required for post processing if you run it in debug mode right now

Comment (riedgar-ms) [reply requirements.txt]: Anything else I should add in?

Comment (romanlutz) [reply requirements.txt]: I don't think anything's missing. pip install has no issues with comments this way, right?

Comment (riedgar-ms) [reply requirements.txt]: I've tested on a fresh environment :-)

Comment (riedgar-ms) [reply requirements.txt]: Are you planning on keeping the matplotlib dependency for postprocessing?

Comment (romanlutz) [reply requirements.txt]: I thought about this a little, and I think the plots are useful. Also, 99.9% of people who do machine learning will have matplotlib installed anyway. I don't think it's unreasonable. 

FYI I've made a note about discussing the package versioning with the entire group in the next meeting.

**Pull request merged**